pkg/flowkit/config/json/account.go

```diff
@@ -166,7 +166,7 @@ func transformAccountsToJSON(accounts config.Accounts) jsonAccounts {
 	for _, a := range accounts {
 		if a.FromFile != "" {
 			jsonAccounts[a.Name] = transformFromFileAccountToJSON(a)
-		} else if isDefaultKeyFormat(a.Key) {
+		} else if isDefaultKeyFormat(a.Key) && !a.SaveAdvancedFormat {
```

Comment:
I would also update the `transformAdvancedToConfig` function to set the new `SaveAdvancedFormat` field to true. This way we prevent the advanced format JSON objects from being converted back to simple format.

Reply:
Good point. Doing that.